app/controllers/api/v2/smart_proxies_controller.rb

```diff
@@ -65,6 +65,17 @@ def refresh
         process_response @smart_proxy.refresh.blank? && @smart_proxy.save && @smart_proxy.reload
       end
 
+      api :POST, "/smart_proxies/:id/import_subnets", N_("Import subnets from Smart proxy")
+      param :id, String, :required => true
+
+      def import_subnets
+        proxy = SmartProxy.find(params[:id])
+        @subnets = Subnet::Ipv4.import(proxy)
+        if @subnets.empty?
```

Comment:
Looking at the source, it can return `nil` and I think `nil.empty?` raises a `NoMethodError`. If it's `nil`, the Smart Proxy does not have the DHCP feature. That may be an `unprocessable_entity`, but I think right now it will result in an internal server error.

Reply:
From what I have checked when `@subnets` is nil the `@subnets.empty?` doesn't raise a `NoMethodError`, can you verify me?

Reply:
Still, `nil` indicates a bad request since an action that can't be performed was requested. This warrants a different response to the client. So in that case I would expect HTTP 422 Unprocessable Entity.